src/index.ts

```diff
@@ -423,12 +514,65 @@ export class FlashbotsBundleProvider extends providers.JsonRpcProvider {
     })
   }
 
+  private waitForTxInclusion(transactionHash: string, startBlockNumber: number, timeout: number, maxBlockNumber?: number) {
+    return new Promise<FlashbotsTransactionResolution>((resolve, reject) => {
+      let timer: NodeJS.Timer | null = null
+      let done = false
+      const maxBlockNum = maxBlockNumber || startBlockNumber + PRIVATE_TX_WAIT_BLOCKS;
+
+      // runs on new block event
+      const handler = async (blockNumber: number) => {
+        if (blockNumber <= maxBlockNum) {
+          // search for tx in block
+          const block = await this.genericProvider.getBlock(blockNumber);
```

Comment:
probably better could be to use `getTransactionByHash`, since a block subscription might skip a block or even two (which i've seen in practice), which would not trigger this code.

Reply:
:100: good idea. Done.